core/corehttp/hostname.go

```diff
@@ -233,21 +233,19 @@ type wildcardHost struct {
 func prepareKnownGateways(publicGateways map[string]*config.GatewaySpec) gatewayHosts {
 	var hosts gatewayHosts
 
-	if len(publicGateways) == 0 {
-		hosts.exact = make(
-			map[string]*config.GatewaySpec,
-			len(defaultKnownGateways),
-		)
-		for hostname, gw := range defaultKnownGateways {
-			hosts.exact[hostname] = gw
-		}
-		return hosts
-	}
```

Comment:
@MichaelMure FYI this is why sharness tests were failing: you were applying implicit defaults only when `Gateway.PublicGateways` was empty. This meant subdomain gateway at `localhost` was missing when you had any public gateway defined. I fixed it in 2ff6f1a and added explicit test to guard against regressions.